### DISCUSSION TITLE
Release a new version?

### DISCUSSION THREAD

**Discussion by tisonkun** [jni-rs/jni-rs#494] [General]
I noticed that there is several changes (even breaking changes) made before the last release (0.21.1).

I suggest we do a release to enable downstream smoothly do the migration instead of collecting more changes which make migration more difficult.

FYI - I'm using jni-rs for [Apache OpenDAL's Java binding](https://github.com/apache/incubator-opendal/blob/main/bindings/java/README.md) and thank a lot for this excellent crate!

**Comment by tisonkun** [comment]
cc @rib @madadam @argv-minus-one what do you think?

**Reply by argv-minus-one** [reply]
Not my call. I believe it's @rib who decides when releases get released.

**Reply by rib** [reply]
yeah I'd like to do a release soonish - has just been a matter of finding some time recently.

I think the jni-sys release is ready to go, and I tried to reach out to the main projects that currently depend on jni-sys to make them aware of the breaking change there.

The jni-sys release is the thing that blocks landing https://github.com/jni-rs/jni-rs/pull/478 (plus some follow up I need to do based on feedback). This is probably the main change I'm keen to have in the next jni-rs release.

**Reply by rib** [reply]
Okey, I've at least published a new `jni-sys` 0.4 release today: https://github.com/jni-rs/jni-sys/releases/tag/v0.4.0 and that can hopefully help unblock preparing a jni crate release.

**Comment by rib** [comment]
It took a long time, but I've just release jni 0.22 so I guess we can close this now: https://github.com/jni-rs/jni-rs/pull/650
